libcudacxx/include/cuda/std/__cuda/cmath_nvbf16.h

```diff
@@ -37,47 +37,47 @@ _LIBCUDACXX_BEGIN_NAMESPACE_STD
 // trigonometric functions
 inline _LIBCUDACXX_INLINE_VISIBILITY __nv_bfloat16 sin(__nv_bfloat16 __v)
 {
-  NV_IF_ELSE_TARGET(NV_IS_DEVICE, (return ::hsin(__v);), (return __nv_bfloat16(::sin(float(__v)));))
+  NV_IF_ELSE_TARGET(NV_IS_DEVICE, (return ::hsin(__v);), (return __float2bfloat16(::sin(__bfloat162float(__v)));))
```

Comment:
Important: The host path does not seem right. The code calls the C library's `::sin` here, which promotes the argumet to a `double`. We should either call `::sinf` or `std::sin` here.

Applies to several math functions below.

Reply:
Created issue: #2078